pkg/controller/iotconfig/jdbc_device_connection.go

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright 2019-2020, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package iotconfig
+
+import (
+	"context"
+	"github.com/enmasseproject/enmasse/pkg/util"
+	"github.com/enmasseproject/enmasse/pkg/util/ext"
+	"strconv"
+
+	iotv1alpha1 "github.com/enmasseproject/enmasse/pkg/apis/iot/v1alpha1"
+	"github.com/enmasseproject/enmasse/pkg/util/cchange"
+	"github.com/enmasseproject/enmasse/pkg/util/install"
+	"github.com/enmasseproject/enmasse/pkg/util/recon"
+	"github.com/ghodss/yaml"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func (r *ReconcileIoTConfig) processJdbcDeviceConnection(ctx context.Context, config *iotv1alpha1.IoTConfig) (reconcile.Result, error) {
+
+	rc := &recon.ReconcileContext{}
+
+	change := cchange.NewRecorder()
+
+	// configmap
+
+	rc.ProcessSimple(func() error {
+		return r.processConfigMap(ctx, nameDeviceConnection+"-config", config, false, func(config *iotv1alpha1.IoTConfig, configMap *corev1.ConfigMap) error {
+			return r.reconcileJdbcDeviceConnectionConfigMap(config, configMap, change)
+		})
+	})
+
+	// deployment
+
+	rc.ProcessSimple(func() error {
+		return r.processDeployment(ctx, nameDeviceConnection, config, false, func(config *iotv1alpha1.IoTConfig, deployment *appsv1.Deployment) error {
+			return r.reconcileJdbcDeviceConnectionDeployment(config, deployment, change)
+		})
+	})
+
+	// done
+
+	return rc.Result()
+}
+
+func (r *ReconcileIoTConfig) reconcileJdbcDeviceConnectionDeployment(config *iotv1alpha1.IoTConfig, deployment *appsv1.Deployment, change *cchange.ConfigChangeRecorder) error {
+
+	install.ApplyDeploymentDefaults(deployment, "iot", deployment.Name)
+	deployment.Annotations[DeviceConnectionTypeAnnotation] = "jdbc"
+	deployment.Annotations[util.ConnectsTo] = "iot-auth-service"
+	deployment.Spec.Template.Spec.ServiceAccountName = "iot-device-connection"
+	deployment.Spec.Template.Annotations[DeviceConnectionTypeAnnotation] = "jdbc"
+
+	service := config.Spec.ServicesConfig.DeviceConnection
+	applyDefaultDeploymentConfig(deployment, service.JDBC.ServiceConfig, change)
+
+	var tracingContainer *corev1.Container
+	err := install.ApplyDeploymentContainerWithError(deployment, "device-connection", func(container *corev1.Container) error {
+
+		tracingContainer = container
+
+		// we indeed re-use the device registry image here
+		if err := install.SetContainerImage(container, "iot-device-registry-jdbc", config); err != nil {
+			return err
+		}
+
+		container.Args = nil
+
+		// set default resource limits
+
+		container.Resources = corev1.ResourceRequirements{
+			Limits: corev1.ResourceList{
+				corev1.ResourceMemory: *resource.NewQuantity(512*1024*1024 /* 512Mi */, resource.BinarySI),
```

Comment:
Should this be overridable somewhere?

Reply:
This is just the default, the override takes place a bit further down below: https://github.com/EnMasseProject/enmasse/pull/4142/files/55e756995990b50ebef0dcb2d0826e7b736969c6#diff-b1e319091064da1010902dc38f8a3c7dR134 

~~~go
// apply container options

if service.JDBC != nil {
  applyContainerConfig(container, service.JDBC.Container)
}
~~~